models/cost_per_serverless_task_run.sql

```diff
@@ -0,0 +1,151 @@
+{{ config(
+    materialized='incremental',
+    unique_key=['query_id', 'start_time'],
+) }}
+
+with
+stop_threshold as (
+    select max(end_time) as latest_ts
+    from {{ ref('stg_serverless_task_history') }}
+),
+
+// task_history has all runs, but we only want serverless runs
+serverless_runs as (
+    select
+        task_history.* exclude (query_id), // just to remove ambiguous column names
+        query_history.*,
+        task_history.database_name as task_database_name,
+        task_history.schema_name as task_schema_name,
+        name as task_name,
+        concat(task_database_name, '.', task_schema_name, '.', task_name) as task_fqn_name,
+        start_time as execution_start_time,
+    from {{ ref('stg_task_history') }} as task_history
+
+    inner join {{ ref('stg_query_history') }} as query_history
+        on task_history.query_id = query_history.query_id and query_history.start_time = task_history.query_start_time
+
+    where true
+        and task_schema_id is not null
+        and task_database_id is not null
+        and startswith(warehouse_name, 'COMPUTE_SERVICE_WH_USER_TASKS') -- TODO: Verify on a larger sample
+        and end_time <= (select latest_ts from stop_threshold)
+        {% if is_incremental() %}
+        -- account for late arriving queries
+        and end_time > (select coalesce(dateadd(day, -3, max(end_time)), '1970-01-01') from {{ this }})
+        {% endif %}
+
+    order by scheduled_time desc
```

Comment:
styling nitpick, but can you please prefix the columns with the table they came from, i.e.

```sql
and task_history.task_database_id is not null)
and startswith(query_history.warehouse_name, 'COMPUTE_SERVICE_WH_USER_TASKS')
...
order by task_history.scheduled_time desc
```

Reply:
oh also, i don't htink we need this sort in here?

Reply:
I agree with the style comment. The `order by` was something I forgot while working on the query on Snowsigth and then moving here to the model. I just removed it, thanks!